wlauto/workloads/octane2/__init__.py

```diff
@@ -0,0 +1,176 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author: Milosz Wasilewski <milosz.wasilewski@linaro.org>
+# pylint: disable=E1101,W0201
+import os
+import time
+import urllib
+import shutil
+import re
+import zipfile
+
+from wlauto import (
+    settings,
+    Parameter,
+    UiAutomatorWorkload
+)
+from wlauto.exceptions import WorkloadError
+from wlauto.utils.types import boolean
+
+# Use master from github as there is no better candidate
+DEFAULT_OCTANE_ARCHIVE = "https://github.com/chromium/octane/archive/master.zip"
+DOWNLOADED_FILE_NAME = "octane2.zip"
+LOGCAT_FILE_NAME = "browser_octane2_logcat.txt"
+
+class Octane2(UiAutomatorWorkload):
+
+    name = 'octane2'
+    description = """
+    Octane 2 is a JavaScript benchmark for browser
+
+    """
+    octane_jar_file = '/org.linaro.wlauto.uiauto.octane2.jar'
+    octane_run_string = 'org.linaro.wlauto.uiauto.octane2.UiAutomation#runUiAutomation'
+    octane_extract_string = 'org.linaro.wlauto.uiauto.octane2.UiAutomation#runUiAutomation'
+
+    parameters = [
+        Parameter('force_dependency_push', kind=boolean, default=False,
+                  description=('Specifies whether to push dependency files to the device to the device '
+                               'if they are already on it.')),
+        Parameter('perform_cleanup', kind=boolean, default=False,
+                  description='If ``True``, workload files on the device will be deleted after execution.'),
+        Parameter('browser_package', default='org.chromium.chrome',
+                  description='Specifies the package name of the device\'s browser app.'),
+        Parameter('browser_activity', default='.browser.ChromeTabbedActivity',
+                  description='Specifies the startup activity  name of the device\'s browser app.'),
+        Parameter('octane_timeout', kind=int, default=900,
+                  description='Specifies the timeout (in seconds) before the uiautomator script is terminated.'),
+        Parameter('clear_file_cache', kind=boolean, default=True,
+                  description='Clear the the file cache on the target device prior to running the workload.'),
+    ]
+
+    supported_platforms = ['android']
+
+    def setup(self, context):  # NOQA
+        UiAutomatorWorkload.setup(self, context)
+        self.octane2_on_device = '/'.join([self.device.working_directory, self.name])
+        self.index_octane = 'file:///{}'.format(self.octane2_on_device) + '/index.html'
```

Comment:
Good point. It actually doesn't show up in the browser (auto correction?) but can be spotted from the log. Fixed.